File: cpp/patterns/pattern8.cpp
/*
Count up until the pattern iteration completes
                    
1                   
2  3                
4  5  6             
7  8  9  10          
11  12  13  14  15       

*/

#include<iostream>
using namespace std;

int main(){
    int i, j, N;
    cin >> N;

    i = 1; 
    int count = 1;

    while(i <= N){

        j = 1;
        while(j <= i){
            cout<<count<<"   ";

            //Counting up as long as this loop continues
            count+=1;
            j++;
        }

        cout<<endl;
        i++;
    }

    return 0;
}

File: js/patterns/pattern8.js
/*
Count up until the pattern iteration completes
                    
1                   
2  3                
4  5  6             
7  8  9  10          
11  12  13  14  15
 */


//Main Function
const main = ()=>{
    let i, j;
    const N = Number(window.prompt('Enter your rows'));
    let string = ''
    let count = 1;

    for(i = 1; i <= N; i++){
        for(j = 1; j <= i; j++){
            string += `${count} `;
            count++;
        }
        string += '\n';
    }
    console.log(string)
}
main();